File: src/lessons/welcome/LoopDoWhileEntity.java
package lessons.welcome;

import java.awt.Color;

public class LoopDoWhileEntity extends jlm.universe.bugglequest.SimpleBuggle {
	boolean isGroundWhite() { 
		return getGroundColor().equals(Color.white)?true:false;
	}
	@Override
	public void run() { 
		/* BEGIN SOLUTION */
		do {
			forward();
		} while (!isGroundWhite());
		/* END TEMPLATE */
	}
}

File: resources/exercises/loopdowhile/LoopDoWhileEntity.c
//RemoteBuggle

int isGroundWhite() {
	return getGroundColor()==white;
}

void run(){
	/* BEGIN TEMPLATE */
	/* BEGIN SOLUTION */
	do {
		forward(1);
	} while (!isGroundWhite());
	/* END SOLUTION */
	/* END TEMPLATE */
}

